Userland/Libraries/LibGfx/PGMLoader.cpp

```diff
@@ -28,25 +25,29 @@ static void set_adjusted_pixels(PGMLoadingContext& context, Vector<Gfx::Color> c
     }
 }
 
-bool read_image_data(PGMLoadingContext& context, Streamer& streamer)
+bool read_image_data(PGMLoadingContext& context)
 {
+    auto& stream = *context.stream;
     Vector<Gfx::Color> color_data;
 
     if (context.type == PGMLoadingContext::Type::ASCII) {
         while (true) {
-            auto number_or_error = read_number(streamer);
+            auto number_or_error = read_number(stream);
             if (number_or_error.is_error())
                 break;
             auto value = number_or_error.value();
 
-            if (read_whitespace(context, streamer).is_error())
+            if (read_whitespace(context).is_error())
                 break;
 
             color_data.append({ (u8)value, (u8)value, (u8)value });
         }
     } else if (context.type == PGMLoadingContext::Type::RAWBITS) {
-        u8 pixel;
-        while (streamer.read(pixel)) {
+        while (true) {
```

Comment:
> 1 byte reads are the minimum any stream can do

That's not true, a `read()` of 0 is perfectly normal when reaching EOF. But it returns an error in case of `read_value` as it can't return a value. And `stream.is_eof()` won't be true until `read` returns a zero. 
In other words, `stream.read_value` will generate an error before we stop iterating.

I refactored the loop to avoid the issue.